metadata-ingestion/src/datahub/ingestion/run/pipeline.py

```diff
@@ -390,7 +390,15 @@ def run(self) -> None:
                         record_envelopes = self.extractor.get_records(wu)
                         for record_envelope in self.transform(record_envelopes):
                             if not self.dry_run:
-                                self.sink.write_record_async(record_envelope, callback)
+                                try:
+                                    self.sink.write_record_async(
+                                        record_envelope, callback
+                                    )
+                                except Exception as e:
+                                    # In case the sink's error handling is bad, we still want to report the error.
+                                    self.sink.report.report_failure(
+                                        f"Failed to write record: {e}"
```

Comment:
Can log stack trace? I'm a bit concerned about losing information where the error came.

Reply:
this is mainly a fallback mechanism if the source doesn't catch the error elsewhere - given that the error handling in the kafka sink has also improved, I think it's ok